### ISSUE TITLE
Issue with enum comparison witch are changed in another function

### ISSUE THREAD

**Issue by ghcr** [microsoft/TypeScript#37075]
**Tested TypeScript Versions**: 3.7.5 & 3.8.2 & Nighly

I got an Error while compiling the following code
```typescript
enum Test {
    SomeItem,
    OtherItem
}

let test = Test.SomeItem

function edit() {
    test = Test.OtherItem
}
edit()

if(test === Test.OtherItem) console.log("Hello World")
// This condition will always return 'false'
// since the types 'Test.SomeItem' and 'Test.OtherItem' have no overlap.
// ts(2367)
```

**Expected behavior**: No errors will be thrown.
`if(test === Test.OtherItem)` is possible because `test` is changed in the `edit` function and the `edit` function is changing `test`.

**Actual behavior**:
TypeScript is throwing the following Error:
`This condition will always return 'false' since the types 'Test.SomeItem' and 'Test.OtherItem' have no overlap. ts(2367)`

**Playground**: https://www.typescriptlang.org/play/index.html#code/KYOwrgtgBAKsDOAXKBvAUASAMoHsLAElFgIAaTAeUQAtgAnIktAXzTQBthlikoBeWAkQA6XPkYQ2AMzAgAxogCWOEFGAATRYgAUASlSYeyAXCTCqtBsUmsNWvW0VTtR-nxNDzNehP1yV8Dicwuw4AObaAEQAEsDsoVAA6jh07OqRumgA9Fmw1IrwUP4gmkoqUADuivFQAIbsFbUAnoV0XGB0qgDkUvXwwF3ZufCK8sBQ3hNNAA4IUF2mImKE1l11JfOLXpYSa9S1AG7jIDhQOEeptdPCQxPw2gBMAMwAbADsukA

**Comment by MartinJohns** [comment]
Yet another duplicate of #9998.
